healthcareai/deploy_supervised_model.py

```diff
@@ -23,150 +25,152 @@ class DeploySupervisedModel(object):
 
         if debug:
             print('\nDataframe after removing DTS columns:')
-            print(df.head())
+            print(self.df.head())
             print('\nNow either doing imputation or dropping rows with NULLs')
             print('\nTotal df shape before impute/remove')
-            print(df.shape)
+            print(self.df.shape)
 
         pd.options.mode.chained_assignment = None  # default='warn'
 
 
         # TODO: put try/catch here when type = class and pred col is numer
-        df.replace('NULL', np.nan, inplace=True)
+        self.df.replace('NULL', np.nan, inplace=True)
 
         if debug:
             print('\nTotal df after replacing NULL with nan')
-            print(df.shape)
-            print(df.head())
+            print(self.df.shape)
+            print(self.df.head())
 
         if self.impute:
 
-            df = DataFrameImputer().fit_transform(df)
+            self.df = DataFrameImputer().fit_transform(self.df)
 
             if debug:
                 print('\nTotal df after doing imputation:')
-                print(df.shape)
-                print(df.head())
+                print(self.df.shape)
+                print(self.df.head())
 
         else:
             # TODO switch similar statements to work inplace
             # If not impute, only remove rows in train that contain NULLS
 
             if debug:
                 print('\nTrain portion before impute/remove')
-                print(df.ix[df[windowcol] == 'N'].shape)
+                print(self.df.ix[self.df[window_column] == 'N'].shape)
 
-            df_train_temp = df.ix[df[windowcol] == 'N']
-            df_test_temp = df.ix[df[windowcol] == 'Y']
+            df_train_temp = self.df.ix[self.df[window_column] == 'N']
+            df_test_temp = self.df.ix[self.df[window_column] == 'Y']
             df_train_temp.dropna(axis=0, how='any', inplace=True)
 
             # Recombine same (unaltered) test rows with modified training set
-            df = pd.concat([df_test_temp, df_train_temp])
+            self.df = pd.concat([df_test_temp, df_train_temp])
 
             if debug:
                 print('\nTotal df after dropping rows with NULLS:')
-                print(df.shape)
+                print(self.df.shape)
                 # noinspection PyUnresolvedReferences
-                print(df.head())
+                print(self.df.head())
                 print('\nTrain portion after removing training rows w/ NULLs')
                 # noinspection PyUnresolvedReferences,PyUnresolvedReferences
-                print(df.ix[df[windowcol] == 'N'].shape)
+                print(self.df.ix[self.df[window_column] == 'N'].shape)
 
         #Call new function!!
         # Convert predicted col to 0/1 (otherwise won't work with GridSearchCV)
         # Makes it so that healthcareai only handles N/Y in pred column
-        if modeltype == 'classification':
+        if model_type == 'classification':
```

Comment:
Consider making this an `Enum` if you're ready to require python 3.4+
https://docs.python.org/3/library/enum.html

Again, this is only for you to consider.  It may be more appropriate to eventually use some kind of inheritance scheme to determine whether a model is a classification model or a regression model (or other).